### ISSUE TITLE
Eventual-consistency-delay works as timeout for currentEventsByTag query

### ISSUE THREAD

**Issue by belyyv** [akka/akka-persistence-cassandra#117]
I have found that on big amount of events curentEventsByTag query successfully stops after `eventual-consistency-delay` time. For me it looks like bug, because it violates currentEventsByTag semantics. 

In my case I had around 1 million resulting events and it took 5 minutes to read and process them. `eventual-consistency-delay` had default value - 10s. And query stopped after that time, getting in fact only small part of 'current' events. It was not easy to found because I don't see any relation between parameter name and that effect.

Increasing `eventual-consistency-delay` causes all currentEventsByTag queries to wait specified time until completion, so it doesn't solve problem nicely.

I see workaround in making actor requiring that big amount of events persisted and using snapshots to limit amount of events to get in one query, so it will fit in `eventual-consistency-delay` time.

But as for me it's still better to report the issue to save time of someone investigating same problem.

---

I believe issue is related to stop condition in following code:

``` scala
  def isCurrentTimeAfterToOffset(): Boolean =
    toOffset match {
      case None    => false
      case Some(_) => System.currentTimeMillis() > toOffsetTimestamp
    }

  def stopIfDone(): Unit = {
    if (buf.isEmpty && (isToOffsetDone || isCurrentTimeAfterToOffset())) {
      onCompleteThenStop()
    }
  }
```

**Comment by patriknw** [comment]
Interesting, thanks for reporting. Sounds like a bug.

**Comment by patriknw** [comment]
This issue has been open for too long. I'll take a look at it right away.

**Comment by notxcain** [comment]
Thanks, @patriknw

**Comment by patriknw** [comment]
@belyyv and @notxcain I was able to reproduce and fix the problem, see https://github.com/akka/akka-persistence-cassandra/pull/156

Note, that when you use `cassandra-query-journal.delayed-event-timeout` I would still recommend to have a short `eventual-consistency-delay=1s` to avoid unnecessary look backs for missing events. I think that will give better overall performance.

**Comment by notxcain** [comment]
@patriknw thanks for the fix and for the note! Do you plan to release `0.22.1` as soon as #156 gets merged?

**Comment by patriknw** [comment]
yes, I'll probably release 0.23 today

**Comment by notxcain** [comment]
That's great to hear!

**Comment by patriknw** [comment]
@notxcain no release today. I found an unrelated performance problem that must be fixed and included in the release.

**Comment by patriknw** [comment]
Released 0.23. (it was not a performance problem)
